### PR TITLE
Fix/refactor param handling

### PR DESCRIPTION
**Pull request by RickWoltheus**
this pr refactors sending ids to other screens for the taalhuis participants views

### CODE REVIEW

File: client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx [hunk 1]
```diff
@@ -83,16 +86,17 @@ export interface FormModel
         ReadingTestInformationFieldsetModel,
         WritingInformationFieldsetModel {}
 
-export const ParticipantsUpdateView: React.FunctionComponent<Props> = () => {
+export const ParticipantsUpdateView: React.FunctionComponent<Props> = props => {
+    const { routeState } = props
     const { i18n } = useLingui()
-    const history = useHistory()
+    const history = useHistory<ParticipantDetailLocationStateProps>()
```
Comment (Grkmk) [inline client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx:92]: the cast type doesnt seem right. isnt state in `history.location`?

File: client/src/views/Authorized/Participants/taalhuis/Participants/ParticipantsOverviewView.tsx [hunk 2]
```diff
@@ -91,11 +92,16 @@ export const ParticipantsOverviewView: React.FunctionComponent<Props> = () => {
...
-                    participantname: participant.nickName,
-                })}
+            <TableLink<ParticipantDetailLocationStateProps>
+                to={{
+                    pathname: routes.authorized.participants.taalhuis.participants.detail.index,
+                    search: '',
+                    hash: '',
+                    state: {
+                        participantId: participant.id,
+                        participantName: 'TEMP name',
```
Comment (Grkmk) [inline client/src/views/Authorized/Participants/taalhuis/Participants/ParticipantsOverviewView.tsx:102]: you should leave a todo comment here

File: client/src/views/Authorized/Participants/taalhuis/Registrations/RegistrationsView.tsx [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
-import { Redirect, Route, Switch } from 'react-router-dom'
+import { Redirect, Route, Switch, useLocation } from 'react-router-dom'
 import { routes } from '../../../../../routes/routes'
 import { RegistrationReadView } from './Detail/RegistrationReadView'
 import { RegistrationsOverviewView } from './RegistrationsOverviewView'
 
 interface Props {}
+export interface RegistrationsDetailLocationStateProps {
+    registrationid: string
+    registrationname: string
```
Comment (Grkmk) [inline client/src/views/Authorized/Participants/taalhuis/Registrations/RegistrationsView.tsx:10]: camelCase, as usual :)

**Review by Grkmk — DISMISSED** [review]
minor stuff

File: client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx (hunk 1 above)
Comment (RickWoltheus) [reply client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx:92]: image
the name suggests it's for the location state, also in the push function i am getting the right typing

Comment (Grkmk) [reply client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx:92]: https://reactrouter.com/web/api/history

Comment (RickWoltheus) [reply client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx:92]: I don't see any docs about this issue. am i overlooking something?

Comment (Grkmk) [reply client/src/views/Authorized/Participants/taalhuis/Participants/Detail/Intake/ParticipantUpdateView.tsx:92]: useHistory hook ([here](https://reactrouter.com/web/api/Hooks/usehistory)) returns a `history` instant, and you can check the history instance properties in the link i sent previously

**Review by bbawuah — APPROVED** [review]

**Pull request merged**